src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -1108,6 +1108,322 @@ void MFEMSidreDataCollection::RegisterField(const std::string& field_name,
   DataCollection::RegisterField(field_name, gf);
 }
 
+// private method
+void MFEMSidreDataCollection::
+addQuadratureFunction(const std::string &field_name, mfem::QuadratureFunction *qf,
```

Comment:
This looks very similar to `addScalarBasedGridFunction` - I wonder if there's an opportunity to combine the two given that both `mfem::GridFunction` and `mfem::QuadratureFunction` inherit from `mfem::Vector`.  It looks like for the most part both of these functions only use the `mfem::Vector` interface.